tests.js

```diff
@@ -3,111 +3,263 @@ var expect = require('expect.js');
+      };
+      var values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
+      expect(values).to.eql([ 123 ]);
     });
-    expect(ccc).to.eql(1);
-    expect(ddd).to.eql(0);
-
-    kibiutils.goToElement(json, [ 'aaa', '1', 'bbb' ], function (nested) {
-      if (nested === 'ccc') {
-        ccc++;
-      } else if (nested === 'ddd') {
-        ddd++;
-      }
+
+    it('value is an array', function () {
+      var json = {
+        aaa: [ 123, 456 ]
+      };
+      var values = kibiutils.getValuesAtPath(json, [ 'aaa' ]);
+      expect(values).to.eql([ 123, 456 ]);
     });
-    expect(ccc).to.eql(1);
-    expect(ddd).to.eql(1);
-  });
 
+    it('nested value is an array', function () {
+      var json = {
+        aaa: {
+          bbb: [ 123, 456 ]
+        }
+      };
+      var values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
+      expect(values).to.eql([ 123, 456 ]);
+    });
 
-  it('goToElement object nested in array', function () {
-    var json = {
-      aaa: [
-        {
-          ccc: {
-            eee: 42
+    it('multiple values are reachable via path', function () {
+      var json = {
+        aaa: [
+          {
+            bbb: 123
+          },
+          {
+            bbb: 456
           }
-        }
-      ]
-    };
-    kibiutils.goToElement(json, [ 'aaa', '0', 'ccc', 'eee' ], function (nested) {
-      expect(nested).to.eql(42);
+        ]
+      };
+      var values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
+      expect(values).to.eql([ 123, 456 ]);
+    });
+
+    it('multiple values with some that are arrays are reachable via path', function () {
+      var json = {
+        aaa: [
+          {
+            bbb: [ 123, 456 ]
+          },
+          {
+            bbb: 789
+          }
+        ]
+      };
+      var values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ]);
+      expect(values).to.eql([ 123, 456, 789 ]);
     });
-  });
 
-  it('goToElement nested object', function () {
-    var json = {
-      aaa: {
-        bbb: [
+    it('nested array mix', function () {
+      var json = {
+        aaa: [
+          {
+            bbb: [
+              {
+                ccc: 123
+              },
+              {
+                ccc: 456
+              }
+            ]
+          },
           {
-            ccc: 'ccc'
+            bbb: {
+              ccc: 789
+            }
           }
         ]
-      }
-    };
+      };
+      var values = kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb', 'ccc' ]);
+      expect(values).to.eql([ 123, 456, 789 ]);
+    });
 
-    kibiutils.goToElement(json, [ 'aaa', 'bbb', '0' ], function (nested) {
-      expect(nested).to.eql({ ccc: 'ccc' });
+    describe('nulls', function () {
+      it('on level 1', function () {
+        var json = {
+          aaa: null
+        };
+        expect(kibiutils.getValuesAtPath(json, [ 'aaa' ])).to.have.length(0);
+        expect(kibiutils.getValuesAtPath(json, [ 'aaa', 'bbb' ])).to.have.length(0);
```

Comment:
Could you also add non-existing path at root like
expect(kibiutils.getValuesAtPath(json, [ 'does-not-exist', 'aaa' ])).to.have.length(0);